### PR TITLE
fix: Switch group widget is crashed when selectedValues is undefined

### LINKED ISSUE #12078: [Bug]: Switch group widget is crashed when selectedValues is undefined
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When `selectedValues` is `undefined`, the app is crashed

### Steps To Reproduce

1. Drop both of switch group widget and checkbox group widget on the canvas.
2. Let the checkbox group widget throw cyclic dependency error by passing a evaluation expression for an option value and clicking on JS toggle in `Options` property.
For example,  BLUE to {{BLUE}}
3. The app is crashed

Reference:
[LOOM DEMO](https://www.loom.com/share/874971cc3ced4f69925b20606e43c6a3)

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by wmdev0808**
## Description
- When selectedValues is undefined, the app is crashed
- Make selected prop for the component to return empty array if its value is falsy

Fixes #12078 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Test A
- Test B

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: fix/12078-switchgroup-selected-undefined 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | app/client/src/widgets/SwitchGroupWidget/component/index.tsx | 26.67 **(0)** | 11.76 **(-0.74)** | 0 **(0)** | 36.36 **(0)**</details>

### CODE REVIEW

File: app/client/src/widgets/SwitchGroupWidget/component/index.tsx [hunk 1]
```diff
@@ -101,7 +101,7 @@ function SwitchGroupComponent(props: SwitchGroupComponentProps) {
         options.map((option: OptionProps) => (
           <StyledSwitch
             alignIndicator={alignment}
-            checked={selected.includes(option.value)}
+            checked={(selected || []).includes(option.value)}
```
Comment (riodeuno) [inline app/client/src/widgets/SwitchGroupWidget/component/index.tsx:104]: @wmdev0808 This fix should do it. However, in the interest of never regressing. Let's add a cypress test for this scenario.

**Review by riodeuno — CHANGES_REQUESTED** [review]
The code looks good. Need to add a cypress test to avoid regressions.

**Review by riodeuno — APPROVED** [review]
Looks good to me!

**Comment by YogeshJayaseelan** [comment]
@wmdev0808 @riodeuno Tested this fix and now the switch widget is not crashing but is it okay to get a cyclic dependency error?


https://www.loom.com/share/d19df25d3dab4895af00300ad42133b4

**Comment by wmdev0808** [comment]
> @wmdev0808 @riodeuno Tested this fix and now the switch widget is not crashing but is it okay to get a cyclic dependency error?
> 
> [LOOM DEMO](https://www.loom.com/share/d19df25d3dab4895af00300ad42133b4)


> @wmdev0808 @riodeuno Tested this fix and now the switch widget is not crashing but is it okay to get a cyclic dependency error?
> 
> [LOOM DEMO](https://www.loom.com/share/d19df25d3dab4895af00300ad42133b4)

@YogeshJayaseelan ,
Yes, you don't have to worry about that

**Comment by riodeuno** [comment]
@YogeshJayaseelan The cyclic dependency error is an issue that needs to be dealt with from the perspective of evaluations. We need to triage that separately. This fix helps prevent the crash of the Switch Group widget.

**Comment by somangshu** [comment]
@YogeshJayaseelan based on the above comment, Can you create another issue around the cyclic dependency and tag the @appsmithorg/fe-coders-pod  for the same. We can also do this when we verify this on release

**Comment by riodeuno** [comment]
@somangshu I think the original issue reported by @YogeshJayaseelan, already talks about the cyclic dependency. We can potentially move that forward for triaging.

**Pull request merged**